core/gui/src/app/dashboard/service/user/dataset/dataset.service.ts

```diff
@@ -61,9 +61,19 @@ export class DatasetService {
     });
   }
 
-  public retrieveDatasetVersionZip(path: string): Observable<Blob> {
-    const encodedPath = encodeURIComponent(path);
-    return this.http.get(`${AppSettings.getApiEndpoint()}/${DATASET_BASE_URL}/version-zip?path=${encodedPath}`, {
+  public retrieveDatasetZip(options: { path?: string; did?: number }): Observable<Blob> {
```

Comment:
add comment here for what this parameter means and how to set them for different purposes.